### PR TITLE
Mobile close dialog

### LINKED ISSUE #913: close dialog on mobile and show Filter(N) button
On mobile we should show a dialog before closing panel letting the user know that their pending changes will be deleted. 
Additionally, we should show number N next to the Fitler button to show how many filters have been applied.

### PR DESCRIPTION
**Pull request by edwingl10**
## Motivation and Context

Fixes #913 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] User visible change (users will notice UI or functional changes)

## How Has This Been Tested?

- [ ] Existing or new tests cover my changes.
- [ ] Manually tested locally or on the Render PR Server.

## Previewing Changes

1. Click the `onrender.com` URL the **render `bot`** posted below.
2. go to scholarships page and apply some filters
3. notice the Filters button shows the number of fitlers applied
4. try closing panel without applying changes

### CODE REVIEW

File: src/components/FilterPanel.jsx
Comment (josueetcom) [inline src/components/FilterPanel.jsx:166]:
```suggestion
            Are you sure you want to close? Your changes will not be saved.
```

File: src/components/FilterPanel.jsx
Comment (josueetcom) [inline src/components/FilterPanel.jsx:163]: Unsaved Changes
```suggestion
        <DialogTitle>Unsaved Changes</DialogTitle>
```

File: src/pages/ListScholarships.jsx
Comment (josueetcom) [inline src/pages/ListScholarships.jsx:25]: This doesn't pick up filters already specified in the URL. E.g. no number appears if you go to https://s4us-pr-914.onrender.com/scholarships?maxAmount=300.

Just use `useQueryParams` in `FilterBar` and calculate the number of filters applied from there.

**Review by josueetcom — CHANGES_REQUESTED** [review]

File: src/components/FilterBar.jsx
Comment (josueetcom) [inline src/components/FilterBar.jsx:18]: Observation: I'm noticing we're going to have to remember to update this when we add new filters later on. 🤔

**Review by josueetcom — APPROVED** [review]

**Comment by josueetcom** [comment]
Be sure to wait until the base branch gets merged first!

File: src/components/FilterBar.jsx
Comment (edwingl10) [reply src/components/FilterBar.jsx:18]: yeah maybe adding it to `useQueryParams`and exporting it from there would be a good option.

**Pull request merged**
